CanvasPlusPlayground/Services/CanvasService.swift

```diff
@@ -263,23 +274,13 @@ struct CanvasService {
     
     // MARK: Repository actions
     
-    @MainActor
-    func setupRepository() async {
-        repository.modelContainer.mainContext.autosaveEnabled = true
-    }
-    
-    func insert(model: Any) async throws {
-        // if data itself is cacheable -> save, if data is an array of cacheables -> save each individually
-        if let toCache = model as? (any Cacheable) {
-            try await repository.insert(toCache)
-        } else if let arrayOfCacheables = model as? [any Cacheable] {
-            for cacheable in arrayOfCacheables {
-                try await repository.insert(cacheable)
-            }
+    func setupRepository() {
```

Comment:
I'd rather this still be an `async` function and use `.task` at the view level. Any reason why you changed this?

Reply:
thought it would be easier for later usage, ill change it back if u think otherwise